sdk/src/main/scala/com/horizen/account/state/AccountStateView.scala

```diff
@@ -225,6 +222,9 @@ class AccountStateView(private val metadataStorageView: AccountStateMetadataStor
         throw new Exception(s"Transaction ${ethTx.id} is invalid.", invalid.getReason)
     }
 
+    // Increase the nonce by 1
+    increaseNonce(message.getFrom.address())
```

Comment:
let's discuss this point.

Reply:
Normally this would have happened in EvmApply but Jan Christoph removed it so that we can control the nonce from outside. 

However, with the old nonce increase position, the nonce was already increased before the transaction got executed - thus the nonce was effectively too high